app/controllers/games_controller.rb

```diff
@@ -42,6 +43,7 @@ def update
   def join
     @game = Game.find_by(id: params[:game_id])
     @game.update_attributes(:black_player_id => current_user.id)
+    @game.make_pieces_for_black_player
```

Comment:
if you use after_save in the games model instead of after_create you can forgo this line entirely

Reply:
i have a branch named something like black player reset if you want a reference